firebase-firestore/src/main/java/com/google/firebase/firestore/model/TargetIndexMatcher.java

```diff
@@ -86,7 +88,11 @@ public class TargetIndexMatcher {
   // The collection ID (or collection group) of the query target.
   private final String collectionId;
 
-  private @Nullable FieldFilter inequalityFilter;
+  // The inequality filters of the target (if it exists).
+  // Note: The sort on FieldFilters is not required. Using SortedSet here just to utilize
+  // the custom comparator.
+  private final SortedSet<FieldFilter> inequalityFilters;
```

Comment:
nit: Use a HashSet if you can since ordering doesn't matter.

Reply:
I need to save the FieldFilters into a set solely based on the Field, so any inequality filters that has same Field but different operators needs to be deduplicated. SortedSet gives me the ability to do custom comparison based on the Field, but HashSet wouldn't, if i remember correctly.

Reply:
HashSet and TreeSet are the same thing. But HashSet uses hashCode() and equals() to determine equality and perform lookups, where TreeSet uses the comparator and stores the elements ascending. So deduplication can be done using either; however, if TreeSet is more convenient because you have the comparator already available then that's fine. But you can definitely achieve the same thing with HashSet. And I'm glad you documented at least that the sort order is not relevant because that will make it much easier in the future if we ever want to swap it out.

Reply:
I am storing `FieldFilter`s only based on its `Field` property, so build in `hashCode()`,`equals()` doesn't meet my need here.
I do not wish to override these functions in the `FiledFilter` class to only consider the `Field` property as it is not a generic case for the class. 
So, giving a custom comparator in TreeSet works nicely here.